scripts/data_bleach.py

```diff
@@ -0,0 +1,64 @@
+import json
+import sys
+
+
+PUBLIC_FIELDS = [x.split('.') for x in [
+    "aliquot.experimental_strategy",
+    "aliquot.node_id",
+    "aliquot.submitter_id",
+    "case.node_id",
+    "case.project_id",
+    "case.submitter_id",
+    "read_group.sequencing_center",
+    "sample.biospecimen_type",
+    "sample.composition",
+    "sample.node_id"]]
+
+
+def error(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+    exit(1)
+
+
+def is_protected_field(key, context):
+    return not any([context + [key] == field for field in PUBLIC_FIELDS])
+
+
+def recursive_clear(d, context=None):
+    """
+    clear the values of a dictionary unless the value is another dictionary
+    in which case recurse
+    """
+    if context is None:
+        context = []
+
+    for key in d:
+        val = d[key]
+        if type(val) == dict:
+            recursive_clear(val, context=context + [key])
```

Comment:
Thanks for doing this Jesse. This recursive search should work for more complex data types as well.